--- xdm/session.c	Mon Mar  2 12:56:36 1998
+++ src/session.c	Sun Sep  5 10:06:22 1999
@@ -500,6 +500,8 @@
     int	pid;
 #ifdef HAS_SETUSERCONTEXT
     struct passwd* pwd;
+    static char  *envinit[1] = { NULL };
+    extern char **environ;
 #endif
 
     if (verify->argv) {
@@ -556,6 +558,8 @@
 	 * Set the user's credentials: uid, gid, groups,
 	 * environment variables, resource limits, and umask.
 	 */
+        /* destroy user environment before calling setusercontext */
+	environ = verify->userEnviron;
 	pwd = getpwnam(name);
 	if (pwd)
 	{
@@ -565,6 +569,7 @@
 			 errno);
 		return(0);
 	    }
+            verify->userEnviron = environ;    
 	    endpwent();
 	}
 	else
